Classes/TextUtils.cs

```diff
@@ -24,19 +24,106 @@ namespace RadioDld
+        *  public static string oldStripDateFromName(string name, DateTime stripDate)
+        *  {
+        *      // Use regex to remove a number of different date formats from episode titles.
+        *      // Will only remove dates with the same month & year as the programme itself, but any day of the month
+        *      // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
+        *      Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+        *      if (matchStripDate.IsMatch(name))
+        *      {
+        *          name = matchStripDate.Match(name).Groups["name"].ToString();
+        *      }
+        *      return name;
+        *  }
+        */
+
         public static string StripDateFromName(string name, DateTime stripDate)
         {
             // Use regex to remove a number of different date formats from episode titles.
-            // Will only remove dates with the same month & year as the programme itself, but any day of the month
-            // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
-            Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+            // Will only remove dates within a set range eitherside of the programme date.
 
-            if (matchStripDate.IsMatch(name))
+            // Issue #59
+            // - Look for date in middle of name (as well as start and end)
+            // - look for other date formats like 13.02.2009; also 13 Feb 2009; 13th feb 2009; 13th-Feb-2009; 13feb09;
+            // - consider full day names as well eg Monday, Tuesday...
+            // - also other delims before and or after the date e.g. 'title - 2009-02-13'
```

Comment:
Again, this can be found in the history on GitHub, so no need for it in the code.